Test/DurableTask.AzureStorage.Tests/TestTablePartitionManager.cs

```diff
@@ -0,0 +1,723 @@
+            {
+                createInstanceTasks[i] = client.CreateOrchestrationInstanceAsync(typeof(HelloOrchestrator), input: i.ToString());
+            }
+            OrchestrationInstance[] instances = await Task.WhenAll(createInstanceTasks);
+
+            var taskHubWorkerTasks = taskHubWorkers.Select(worker => worker.StartAsync());
+            await Task.WhenAll(taskHubWorkerTasks);
+
+            //Check all the workers are not listening to the same queue at the same time during the balancing.
+            bool isBalanced = false;
+            var timeout = TimeSpan.FromSeconds(30);
+            var sw = Stopwatch.StartNew();
+            while (!isBalanced)
+            {
+                Assert.IsTrue(sw.Elapsed <= timeout, "Timeout expired!");
+                var partitions = services[0].ListTableLeases();
+                Assert.AreEqual(4, partitions.Count());
+                isBalanced = (partitions.Count(p => p.CurrentOwner == "0") == 1) &&
+                             (partitions.Count(p => p.CurrentOwner == "1") == 1) &&
+                             (partitions.Count(p => p.CurrentOwner == "2") == 1) &&
+                             (partitions.Count(p => p.CurrentOwner == "3") == 1);
+
+                string[] array0 = services[0].OwnedControlQueues.Select(i => i.Name).ToArray();
+                string[] array1 = services[1].OwnedControlQueues.Select(i => i.Name).ToArray();
+                string[] array2 = services[2].OwnedControlQueues.Select(i => i.Name).ToArray();
+                string[] array3 = services[3].OwnedControlQueues.Select(i => i.Name).ToArray();
+                bool allUnique = CheckAllArraysUnique(array1, array2, array3, array0);
+                await Task.Delay(1000);
+
+                Assert.IsTrue(allUnique, "Multiple workers lsiten to the same queue at the same time.");
+            }
+            
+            //Check all the instances could be processed successfully. 
+            OrchestrationState[] states = await Task.WhenAll(
+                instances.Select(i => client.WaitForOrchestrationAsync(i, TimeSpan.FromSeconds(30))));
+            Assert.IsTrue(
+                Array.TrueForAll(states, s => s?.OrchestrationStatus == OrchestrationStatus.Completed),
+                "Not all orchestrations completed successfully!");
+            
+            var stopServiceTasks = taskHubWorkers.Select(worker => worker.StopAsync());
+            await Task.WhenAll(stopServiceTasks);
+        }
+
+        [KnownType(typeof(Hello))]
+        internal class HelloOrchestrator : TaskOrchestration<string, string>
+        {
+            public override async Task<string> RunTask(OrchestrationContext context, string input)
+            {
+                //  await contextBase.ScheduleTask<string>(typeof(Hello), "world");
+                //   if you pass an empty string it throws an error
+                return await context.ScheduleTask<string>(typeof(Hello), "world");
+            }
```

Comment:
nit: should these comments be removed?
```suggestion
            public override async Task<string> RunTask(OrchestrationContext context, string input)
            {
                return await context.ScheduleTask<string>(typeof(Hello), "world");
            }
```